### PR TITLE
Update phpunit and wp-phpunit packages

### PR DESCRIPTION
**Pull request by shadyvb**
Ref humanmade/product-dev#1034

### CODE REVIEW

**Comment by shadyvb** [comment]
@roborourke I kept running into JS dep conflicts, so opted for using `--legacy-peer-deps` with `npm install`, things seem to work, however not entirely sure about it. Tried to resolve it with no luck. Thoughts ?

**Review by roborourke — COMMENTED** [review]
Some of the `composer.json` edits don't make sense, and I'm not super clear on why any JS changes were necessary but if the dev server etc works fine still then I don't see any problem. If the GH action is using `ubuntu-latest` and we're not specifying the node version that might be the issue. The node version should always be pinned, and preferably the ubuntu version too to keep a consistent dev environment.

File: composer.json [hunk 1]
```diff
@@ -15,24 +15,24 @@
...
     "phpstan/phpstan": "0.12.57",
-    "phpunit/phpunit": "^7",
+    "phpunit/phpunit": "^9.5.20",
     "roots/wordpress": "~5.9.0",
     "squizlabs/php_codesniffer": "3.5.8",
     "szepeviktor/phpstan-wordpress": "0.7.1",
     "vlucas/phpdotenv": "^3",
     "wp-cli/db-command": "^2",
-    "wp-phpunit/wp-phpunit": "~5.9.0",
+    "wp-phpunit/wp-phpunit": "5.9.3",
```
[the remark this answers is not in the document]
Comment (shadyvb) [reply composer.json:24]: Ah, the version constraint was not intentional, should've updated but kept the tilda. Sorry. And yes we should probably upgrade to 6. Will do.

[the remark this answers is not in the document]
Comment (shadyvb) [reply composer.json]: I'm opting to keep on 5.9.3 until WP 6.0 is properly tested. Fixed the tilda.

**Pull request merged**